diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/DiktatRuleSetProvider.kt

```diff
@@ -88,26 +83,19 @@ import org.cqfn.diktat.ruleset.rules.chapter6.classes.SingleConstructorRule
 import org.cqfn.diktat.ruleset.rules.chapter6.classes.SingleInitRule
 import org.cqfn.diktat.ruleset.rules.chapter6.classes.StatelessClassesRule
 
-import com.pinterest.ktlint.core.RuleSet
-import com.pinterest.ktlint.core.RuleSetProvider
 import mu.KotlinLogging
 import org.jetbrains.kotlin.org.jline.utils.Levenshtein
 
 import java.io.File
 
 /**
- * [RuleSetProvider] that provides diKTat ruleset.
- * By default, it is expected to have diktat-analysis.yml configuration in the root folder where 'ktlint' is run
+ * By default, it is expected to have `diktat-analysis.yml` configuration in the root folder where 'ktlint' is run
  * otherwise it will use default configuration where some rules are disabled.
  *
- * The no-argument constructor is used by the Java SPI interface; that's why
- * it's explicitly annotated with [JvmOverloads].
- *
- * @param diktatConfigFile - configuration file where all configurations for inspections and rules are stored
+ * @param diktatConfigFile the configuration file where all configurations for
+ *   inspections and rules are stored.
  */
-class DiktatRuleSetProvider
-@JvmOverloads
-constructor(private var diktatConfigFile: String = DIKTAT_ANALYSIS_CONF) : RuleSetProvider {
+internal class DiktatRuleSetProvider(private var diktatConfigFile: String = DIKTAT_ANALYSIS_CONF) : DiktatRuleSetFactory {
```

Comment:
why diktatConfigFile is `var`?

Reply:
It's old code, I haven't touched it.

The property gets overwritten as `configRules` are initialized, [line 109](https://github.com/saveourtool/diktat/blob/feature/ktlint-wrapper/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/DiktatRuleSetProvider.kt#L109).

Reply:
looks strange, can we extract logic of resolving diktatConfigFile to external method?

Reply:
Surely we can.

Will create a separate PR as a part of #1543 (please don't mark this as resolve before I do).